modules/gmc-agent/src/main/java/org/gridgain/action/controller/BaselineActionsController.java

```diff
@@ -64,15 +65,54 @@ public void updateAutoAdjustAwaitingTime(long awaitingTime) {
     }
 
     /**
-     * @param ids Node ids.
-     * @return Completeble feature.
+     * @param consIds Node consistent ids.
      */
-    public void setBaselineTopology(Collection<String> ids) {
+    public void setBaselineTopology(Collection<String> consIds) {
         authorizeIfNeeded(ctx.security(), SecurityPermission.ADMIN_OPS);
 
-        Set<UUID> uuids = ids.stream().map(UUID::fromString).collect(Collectors.toSet());
+        ctx.grid().cluster().setBaselineTopology(findNodesByConsistentIds(consIds));
+    }
+
+    /**
+     * @param consIds Node consistent ids.
+     */
+    private Collection<BaselineNode> findNodesByConsistentIds(Collection<String> consIds) {
+        Map<String, BaselineNode> baseline = currentBaseLine();
+        Map<String, BaselineNode> srvrs = currentServers();
+
+        Collection<BaselineNode> baselineTop = new ArrayList<>();
+
+        for (String consistentId : consIds) {
+            if (srvrs.containsKey(consistentId))
+                baselineTop.add(srvrs.get(consistentId));
+
+            else if (baseline.containsKey(consistentId))
+                baselineTop.add(baseline.get(consistentId));
 
-        Collection<ClusterNode> nodes = ctx.grid().cluster().forNodeIds(uuids).forServers().nodes();
-        ctx.grid().cluster().setBaselineTopology(nodes);
+            else
+                throw new IgniteIllegalStateException("Check arguments. Node with consistent ID [" + consistentId +
+                    "] not found in server nodes.");
+        }
+
+        return baselineTop;
+    }
+
+    /**
+     * @return Current baseline.
+     */
+    private Map<String, BaselineNode> currentBaseLine() {
+        if (F.isEmpty(ctx.grid().cluster().currentBaselineTopology()))
```

Comment:
Lets extarct to local variable `cluster = ctx.grid().cluster()` ?

Reply:
if we add utility method `fromNullableCollection` this will be not needed

Reply:
Done.